src/main/java/com/example/Restaurant_Management_System_REST_API/service/TableService.java

```diff
@@ -0,0 +1,103 @@
+
+        return tableList;
+    }
+
+    private Table findById(Long id) throws NotFoundInDatabaseException {
+        return tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+    }
+
+    public ResponseEntity<?> deleteById(Long id) throws NotFoundInDatabaseException {
+        Table tableToDelete = tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+
+        tableRepository.delete(tableToDelete);
+
+        return new ResponseEntity<>("Table with id " + tableToDelete.getId() + " has been deleted!", HttpStatus.OK);
+    }
+
+    void iterateAndSetTablesToReservationAndSave(Reservation reservation) throws NotFoundInDatabaseException {
+        for (Table table : reservation.getTables()) {
+            checkIfTableExist(table.getId());
+            List<Reservation> reservationList = new ArrayList<>();
+
+            if(table.getReservationList() != null)
+                reservationList = table.getReservationList();
+
+            reservationList.add(reservation);
+            table.setReservationList(reservationList);
+
+            table.setAvailable(false);
+            tableRepository.save(table);
+        }
+    }
+
+    private void checkIfTableExist(Long id) throws NotFoundInDatabaseException {
+        tableRepository.findById(id).orElseThrow(() -> new NotFoundInDatabaseException(Table.class));
+    }
+
+    void iterateAndSetReservationToNullInTablesAndSave(Reservation reservationToBeDeleted) {
+        for (Table table : reservationToBeDeleted.getTables()) {
+            table.setReservationList(null);
+            table.setAvailable(true);
+            tableRepository.save(table);
+        }
+    }
+
+    void checkIfTablesAreAvailable(Reservation reservation) throws NotFoundInDatabaseException {
+        for (Table table : reservation.getTables()) {
+            Table existingTable = findById(table.getId());
+
+            if (existingTable != null) {
+                for (Reservation existingReservation : existingTable.getReservationList()) {
+                    if (isTimeConflict(existingReservation.getStart(), reservation.getStart())) {
+                        throw new ResponseStatusException(HttpStatus.CONFLICT, "Table with id " + table.getId() +
+                                " is not available at the requested time.");
+                    }
+                }
+            }
+        }
+    }
+
+    boolean isTimeConflict(LocalDateTime existingStart, LocalDateTime newStart) {
+        LocalDateTime existingEnd = existingStart.plusHours(2);
+        LocalDateTime newEnd = newStart.plusHours(2);
+
+        return (newStart.isEqual(existingStart) || newStart.isAfter(existingStart)) && newStart.isBefore(existingEnd)
+                || (newEnd.isAfter(existingStart) && newEnd.isBefore(existingEnd))
+                || (existingStart.isAfter(newStart) && existingStart.isBefore(newEnd));
+    }
```

Comment:
This piece of code here could be described with comments to vastly improve readability. Just by looking at it I'm not sure if it's correctly implemented without testing it out myself. If comments are in place, then I'd have more faith that it is actually tested – and more importantly – that it is understood by other developers, who'll just need to read comments in order to grasp a concept which was in author's mind :)

Reply:
okay! I add some comments! it works properly, I tested it many times different scenarios and that was blowing my mind as well haha

Reply:
That is okay, but I wanted to see a described code and not only the case :) Leave a TODO there and I'll help you with this but can't do that right now.